### PR TITLE
Subscription Discovery for enforcer

### PR DESCRIPTION
**Pull request by sajinieKavindya**
### Purpose

- Fetching subscription data to the adapter at startup
- Adding Subscription Discovery service to send the subscription data to enforcer

### Issues

Fixes https://github.com/wso2/product-microgateway/issues/1591, https://github.com/wso2/product-microgateway/issues/1590

### Automation tests
 - Unit tests added: No
 - Integration tests added: No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [ ] Assigned 'Type' label
- [ ] Assigned the project
- [ ] Validated respective github issues
- [ ] Assigned milestone to the github issue(s)

### CODE REVIEW

File: adapter/pkg/mgw/mgw.go [hunk 1]
```diff
@@ -70,7 +72,7 @@ func init() {
 
 const grpcMaxConcurrentStreams = 1000000
 
-func runManagementServer(server xdsv3.Server, enforcerServer xdsv3.Server, port uint) {
+func runManagementServer(server xdsv3.Server, enforcerServer xdsv3.Server, enforcerSdsServer xdsv3.Server, enforcerAppDsSrv xdsv3.Server, enforcerAPIDsSrv xdsv3.Server, enforcerAppPolicyDsSrv xdsv3.Server, enforcerSubPolicyDsSrv xdsv3.Server, enforcerAppKeyMappingDsSrv xdsv3.Server, port uint) {
```
Comment (VirajSalaka) [inline adapter/pkg/mgw/mgw.go:75]: line contains too much characters.

File: adapter/pkg/messaging/eventtypes.go [hunk 2]
```diff
@@ -67,7 +67,7 @@ type Event struct {
 
 // APIEvent for struct API events
 type APIEvent struct {
-	APIID         string   `json:"apiId"`
+	APIID         int32   `json:"apiId"`
```
Comment (dinusha92) [inline adapter/pkg/messaging/eventtypes.go:70]: This should be a UUID. Hence the type is string. I scanned through the code and noticed that the event usage of APIID is being parsed into string as well. Hence no point of keeping this as int.

Comment (dinusha92) [reply adapter/pkg/messaging/eventtypes.go:70]: Then revert the changes in notificationListner.go

Comment (sajinieKavindya) [reply adapter/pkg/messaging/eventtypes.go]: Done

File: adapter/pkg/mgw/mgw.go (hunk 1 above)
Comment (sajinieKavindya) [reply adapter/pkg/mgw/mgw.go]: fixed

**Review by VirajSalaka — APPROVED** [review]

**Review by dinusha92 — APPROVED** [review]

**Pull request merged**